CMakeLists.txt

```diff
@@ -1,8 +1,6 @@
-cmake_minimum_required(VERSION 3.5.1)
+cmake_minimum_required(VERSION 3.5.1...3.29.0)
 if(CMAKE_VERSION VERSION_LESS 3.12)
     cmake_policy(VERSION ${CMAKE_VERSION})
```

Comment:
From: https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html

"If the running version of CMake is older than 3.12, the extra ... dots will be seen as version component separators, resulting in the ...<max> part being ignored and preserving the pre-3.12 behavior of basing policies on <min>."

Reply:
Also, it might make sense to start requiring CMake 3.12 as a minimum? Up to maintainers here.

Reply:
We test zlib-ng on wide range of Linux versions and some of them don't have cmake 3.12... We avoid adding unnecessary binary packages or recompiling from sources as the current CI workflows are already pushing limits of the runners and any unnecessary additions would cause unnecessary failure. People say building newer cmake works if glibc is at least some version without actually checking what glibc version is used on each distribution and release... I've seen more than once that the bundled version is one less than what people assume it would be.

It is out of scope of each PR to repeat all the unfixed bugs in new releases when we have already downgraded CI instances to use older releases that actually can build zlib-ng successfully.

Reply:
This change is compatible with versions of cmake before 3.12. Jus to be clear.